workers/unity/Packages/io.improbable.gdk.mobile/Editor/MobileLauncherWindow.cs

```diff
@@ -48,112 +53,181 @@ public void OnGUI()
+                        noDevicesFound
+                            ? EditorGUILayout.Popup(prettyDeviceType, 0, emptyDeviceNameList)
+                            : EditorGUILayout.Popup(prettyDeviceType, index, names));
                 }
 
-                using (new EditorGUI.DisabledScope(simulatorNames.Length == 0))
+                var buttonIcon = new GUIContent(EditorGUIUtility.IconContent("Refresh"))
                 {
-                    if (GUILayout.Button("Launch iOS app in Simulator"))
-                    {
-                        var availableSimulators = iOSLaunchUtils.RetrieveAvailableiOSSimulators();
-                        if (availableSimulators.TryGetValue(simulatorNames[simulatorNameIndex], out var simulatorUID))
-                        {
-                            iOSLaunchUtils.Launch(launchConfig.ShouldConnectLocally, simulatorUID, launchConfig.RuntimeIp, true);
-                        }
-                        else
-                        {
-                            simulatorNames = availableSimulators.Keys.ToArray();
-                            simulatorNameIndex = 0;
-                            Debug.LogError("Failed to launch app on selected simulator.");
-                        }
-                    }
-                }
+                    tooltip = $"Refresh your {prettyDeviceType} list."
+                };
 
-                CommonUIElements.DrawHorizontalLine(8, DarkGrey);
+                if (GUILayout.Button(buttonIcon, EditorStyles.miniButton, GUILayout.ExpandWidth(false)))
+                {
+                    RefreshEmulatorsAndDevices(deviceType);
+                }
+            }
 
-                using (new GUILayout.HorizontalScope())
+            // Draw a launch button that will launch the app on the correct platform and chosen device
+            using (new EditorGUI.DisabledScope(noDevicesFound))
+            {
+                if (!GUILayout.Button($"Launch app on {prettyDeviceType}"))
                 {
-                    using (new EditorGUI.DisabledScope(deviceNames.Length == 0))
-                    {
-                        deviceNameIndex = EditorGUILayout.Popup("Device Model", deviceNameIndex, deviceNames);
-                    }
-
-                    var buttonIcon = new GUIContent(EditorGUIUtility.IconContent("Refresh"))
-                    {
-                        tooltip = "Refresh your device list."
-                    };
-
-                    if (GUILayout.Button(buttonIcon, EditorStyles.miniButton, GUILayout.ExpandWidth(false)))
-                    {
-                        deviceNames = iOSLaunchUtils.RetrieveAvailableiOSDevices().Keys.ToArray();
-                        deviceNameIndex = 0;
-                    }
+                    return;
                 }
 
-                using (new EditorGUI.DisabledScope(deviceNames.Length == 0))
+                if (!devices[index].deviceName.Equals(names[index]))
```

Comment:
I think this check is moot and we can call Launch directly but the issue is that the device list might be out of data

we could either:
- refresh the list before calling `launch`, and match on the device of the same name
- let it try to launch and if it fails in either adb/instruments then user can resolve that 

depends on how much we want to verify for ourselves here

Reply:
Under what conditions will this happen?

Reply:
1. connect device
2. refresh list
3. select device
4. disconnect device
5. hit launch button on device that is no longer connected

Reply:
I think we just do:

> let it try to launch and if it fails in either adb/instruments then user can resolve that

as long as we handle and log the errors properly.

Reply:
yeah that's basically what I ended up doing after discussing with Martijn yesterday evening